h/api/test/models_test.py

```diff
@@ -78,6 +79,74 @@ def test_title_annotation_that_has_no_document():
     assert models.Annotation().title == ''
 
 
+@pytest.mark.parametrize("input_,expected", [
+    # If the annotation has both a uri and a document title it should return
+    # the title linked to the uri.
+    (('http://example.com/example.html', 'title'),
+     '<a href="http://example.com/example.html">title</a>'),
+
+    # If the annotation an https uri and a document title it should return the
+    # title linked to the uri.
+    (('https://example.com/example.html', 'title'),
+     '<a href="https://example.com/example.html">title</a>'),
+
+    # If the annotation has a document title but no uri it should just return
+    # the title, not linked.
+    ((None, 'title'),
+     'title'),
+
+    # If the annotation has a document title and a non-http(s) uri it should
+    # just return the title with the filename after it.
+    (('file:///home/bob/Documents/example.pdf', 'title'),
+     'title (example.pdf)'),
+
+    # If the filename is the same as the title it shouldn't show it.
+    (('file:///home/bob/Documents/example.pdf', 'example.pdf'),
+     'example.pdf'),
+
+    # If the annotation has an http(s) uri but no document title it should
+    # return the uri linked to the uri.
+    (('http://example.com/example.html', None),
+     '<a href="http://example.com/example.html">http://example.com/example.html</a>'),
```

Comment:
Given the limited space for the displayed URL, I suggest we should strip uninformative parts such as the scheme and leading 'www.' from what is rendered.